core/services/head_tracker.go

```diff
@@ -496,7 +408,7 @@ func (ht *HeadTracker) receiveHeaders(ctx context.Context) error {
 		select {
 		case <-ht.done:
 			return nil
-		case blockHeader, open := <-ht.outHeaders:
+		case blockHeader, open := <-ht.headers:
 			if !open {
 				return errors.New("HeadTracker: outHeaders prematurely closed")
 			}
```

Comment:
hm i wonder if this closing behaviour should be different now. In the ringbuffer we had:
```
if !ok {
				// If the channel is closed, do not handle any more heads.
				close(r.out)
				return
			}
```
Though it was also closing `r.out` so we would still result in `return errors.New("HeadTracker: outHeaders prematurely closed")` unless `case <-ht.done:` was received first.
Maybe it's fine, but would be good to test this disconnection